overlord/auth/auth.go

```diff
@@ -448,17 +448,17 @@ func (ac *authContext) StoreID(fallback string) (string, error) {
 	return fallback, nil
 }
 
-// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the encoded device serial assertion. It returns ErrNoSerial if the device serial is not yet initialized.
-func (ac *authContext) DeviceSessionRequest(nonce string) (deviceSessionRequest []byte, serial []byte, err error) {
+// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the encoded device serial and model assertions. It returns ErrNoSerial if the device serial is not yet initialized.
+func (ac *authContext) DeviceSessionRequest(nonce string) (deviceSessionRequest []byte, serial []byte, model []byte, err error) {
```

Comment:
this is also very confusing: you have methods that are called the same, that take the same arguments and return the same number of values, that can be nil, but sometimes they're structs and sometimes they are []bytes. Reading code that uses this quickly becomes intractable.